src/hooks/useAIComm.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import { useState } from "react";
+import toast from "react-hot-toast";
+import type { UserAndAIMessage } from "~/server/api/routers/userAndAIComm";
+import { api } from "~/utils/api";
+
+export type AICommResult = [
+  UserAndAIMessage[],
+  () => void,
+  (value: string) => void,
+  currentMessage: string,
+  isAIResponding: boolean,
+];
+
+export default function useAIComm(
+  sessionId: string,
+  userId: string,
+): AICommResult {
+  const [chatState, setChatState] = useState({
+    currentMessage: "",
+    isAIResponding: false,
+  });
+
+  const utils = api.useContext();
+
+  const allSessionMessages =
+    api.userAndAIMessages.getAllSessionUserAndAIMessages.useQuery({
+      sessionId,
+      userId,
+    }).data;
```

Comment:
optional

```suggestion
  const { data: allSessionMessages } =
    api.userAndAIMessages.getAllSessionUserAndAIMessages.useQuery({
      sessionId,
      userId,
    });
```

good thing about follows this convention when using useQuery is that it makes it easy to destructure other returned fields like `isLoading` when required